pkl-core/src/main/java/org/pkl/core/module/ModuleKeys.java

```diff
@@ -494,7 +509,7 @@ public ResolvedModuleKey resolve(SecurityManager securityManager)
         }
         securityManager.checkResolveModule(redirected);
         var text = IoUtils.readString(stream);
-        return ResolvedModuleKeys.virtual(this, uri, text, true);
+        return ResolvedModuleKeys.virtual(this, redirected, text, true);
```

Comment:
I pondered on this for a while because I had to make the same decision a few lines above. I concluded that the use of `uri` here is a bug. It means that `ModuleCache.modulesByResolvedUri` will cache the module under its unresolved URI, which is exactly what `ModuleCache.modulesByOriginalUri` does. So `modulesByResolvedUri` no longer serves any purpose.

Reply:
"resolve" here means resolving the in-language URI to the actual URI used for fetching a module's contents.

For example, an in-language URI might be `projectpackage://pkg.pkl-lang.com/foo@1.0.0#/foo.kl`, whereas the resolved URI might be `file:///path/to/foo.pkl`. It's useful here because the same actual file can be imported using two _different_ URIs in language, but actually are both the same file and we don't need to load it twice.

> It means that ModuleCache.modulesByResolvedUri will cache the module under its unresolved URI, which is exactly what ModuleCache.modulesByOriginalUri does. So modulesByResolvedUri no longer serves any purpose

There's plenty of module keys that have the same entries in both `modulesByResolvedUri` and `modulesByOriginalUri` (e.g. `file:` modules). It's expected for things like this.